snapcraft/models/assertions.py

```diff
@@ -37,25 +37,34 @@ def _to_string(data: Any) -> Any:
 
     Supported scalar types are str, bool, and numbers.
     """
-    # check for a string first, as it is the most common scenario
+    # check for a string first, as it is the most common scenario and prevents
+    # strings being treated as collections of characters.
     if isinstance(data, str):
         return data
 
     if isinstance(data, abc.Mapping):
         return {_to_string(key): _to_string(value) for key, value in data.items()}
 
+    # Check collection *excluding* strings which were handled above.
     if isinstance(data, abc.Collection):
         return [_to_string(i) for i in data]
 
     if isinstance(data, numbers.Number | bool):
         return str(data)
 
+    # Return None as None, otherwise cast_dict_scalars_to_strings might turn it into "None" string
+    if data is None:
+        return None
+
+    # Fallback for unknown types, return as is
     return data
 
 
 class ConfdbSchema(models.CraftBaseModel):
     """Access and data definitions for a specific facet of a snap or system."""
 
+    summary: str | None = pydantic.Field(default=None, description="Optional summary for the specific rule.")
+
```

Comment:
The whole purpose of #5442 was to add a summary to one section of this model and the PR adds it to 3 sections.